src/openzeppelin/token/erc1155/presets/ERC1155MintableBurnable.cairo

```diff
@@ -0,0 +1,252 @@
+        range_check_ptr
+    }(uri: felt):
+    Ownable.assert_only_owner()
+    ERC1155._set_uri(uri)
+    return ()
+end
+
+@external
+func setApprovalForAll{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(operator: felt, approved: felt):
+    ERC1155.set_approval_for_all(operator, approved)
+    return ()
+end
+
+@external
+func safeTransferFrom{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        from_: felt,
+        to: felt,
+        id: Uint256,
+        amount: Uint256,
+        data_len: felt,
+        data: felt*
+    ):
+    ERC1155.safe_transfer_from(from_, to, id, amount, data_len, data)
+    return ()
+end
+
+
+@external
+func safeBatchTransferFrom{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        from_: felt,
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*
+    ):
+    ERC1155.safe_batch_transfer_from(
+        from_, to, ids_len, ids, amounts_len, amounts, data_len, data)
+    return ()
+end
+
+@external
+func mint{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        to: felt,
+        id: Uint256,
+        amount: Uint256,
+        data_len: felt,
+        data: felt*
+    ):
+    Ownable.assert_only_owner()
+    let (caller) = get_caller_address()
+    ERC1155._mint(to, id, amount, data_len, data)
+    return ()
+end
+
+@external
+func mintBatch{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*
+    ):
+    Ownable.assert_only_owner()
+    let (caller) = get_caller_address()
+    ERC1155._mint_batch(to, ids_len, ids, amounts_len, amounts, data_len, data)
+    return ()
+end
+
+@external
+func burn{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(from_: felt, id: Uint256, amount: Uint256):
+    ERC1155.assert_owner_or_approved(owner=from_)
+    let (caller) = get_caller_address()
+    with_attr error_message("ERC1155: called from zero address"):
+        assert_not_zero(caller)
+    end
+    ERC1155._burn(from_, id, amount)
+    return ()
+end
+
+@external
+func burnBatch{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr
+    }(
+        from_: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*
+    ):
+    ERC1155.assert_owner_or_approved(owner=from_)
+    let (caller) = get_caller_address()
+    with_attr error_message("ERC1155: called from zero address"):
+        assert_not_zero(caller)
+    end
```

Comment:
wouldn't `assert_owner_or_approved` always prevent the zero address to reach this point?

Reply:
Yes, the order of checks is wrong here. The test is itself redundant, but maybe helpful in debugging. Should it be dropped?